init/eessi_environment_variables

```diff
@@ -76,12 +76,17 @@ if [ -d $EESSI_PREFIX ]; then
             false
           fi
 
-          export LMOD_RC="$EESSI_SOFTWARE_PATH/.lmod/lmodrc.lua"
-          if [ -f $LMOD_RC ]; then
+          export LMOD_CONFIG_DIR="$EESSI_SOFTWARE_PATH/.lmod"
+          if [ -f $LMOD_CONFIG_DIR/lmodrc.lua ]; then
             show_msg "Found Lmod configuration file at $LMOD_RC"
           else
             error "Lmod configuration file not found at $LMOD_RC"
           fi
+          # Allow for site-specific lmod configuration
+          host_lmod_rc="$EESSI_CVMFS_REPO"/host_injections/.lmod/lmodrc.lua
```

Comment:
I like the idea in principle, but I'd make it sit within a host_injections equivalent of LMOD_CONFIG_DIR

Reply:
You mean: nested, so that you have e.g. a `/cvmfs/software.eessi.io/host_injections/2023.06/software/linux/x86_64/amd/zen2/.lmod/lmodrc.lua` ?

We could. But do you really need to be able to make architecture-specific hooks? In EESSI itself, we only make _one_ hook file with `create_lmodrc.py` and use logic to do architecture specific things like `if (moduleName == "OpenMPI") and (cpuTarget == "aarch64/neoverse_v1") then ...`. Then we just duplicate the `lmodrc.lua` to all prefixes

The downside of having this nested in an architecture prefix is that you might need to do a lot of duplication (or at the very least symlinking) as a host site if you have a heterogenous system. For the Snellius system at SURF, it'd already mean putting the same `lmodrc.lua` in two different prefixes today, and if we have a `zen4` tree tomorrow, I'll also need to _remember_ to deploy it there...